components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -730,18 +730,28 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 					}
 
 					imagestreamFound := false
-					// List imagestreams in the specified namespace
+					// Check for controller namespace before the notebook namespace
+					// to avoid user overriding the image selection in the case of
+					// images having the same name in both namespaces.
+
+					// List imagestreams in the controller namespace
 					imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
-					if err != nil {
-						if k8serr.IsForbidden(err) {
-							log.Info("Permission denied to list imagestreams", "namespace", namespace, "error", err)
-							// fast exit on permission denied
-							return err
-						}
-						log.Info("Cannot list imagestreams", "namespace", namespace, "error", err)
-						continue
+					if err != nil && k8serr.IsForbidden(err) {
+						log.Info("Permission denied to list imagestreams", "namespace", namespace, "error", err)
+						// fast exit on permission denied
+						return err
 					}
 
+					// List imagestreams in the specified notebook namespace
+					inNamespaceImagestreams, err := dynamicClient.Resource(ims).Namespace(notebook.namespace).List(ctx, metav1.ListOptions{})
+					if err != nil && k8serr.IsForbidden(err) {
+						log.Info("Permission denied to list imagestreams", "namespace", notebook.namespace, "error", err)
+						// Permission denied, pass on to controller namespace
+					}
+
+					// Merge the imagestreams from both namespaces
+					imagestreams.Items = append(imagestreams.Items, inNamespaceImagestreams.Items...)
```

Comment:
I'd prefer [`slices.Concat`](https://pkg.go.dev/slices@go1.23.6#Concat), [blog](https://blog.carlana.net/post/2024/golang-slices-concat/) here. But I'm sure that's just aesthetics, and it's just me.

```go
import (
	"fmt"
	"slices"
)

func main() {
	global := []string{"a", "b"}
	local := []string{"c", "d"}
	combined := slices.Concat(global, local)
	fmt.Printf("%+v\n", combined)
}
```